### PR TITLE
Drastically improve detection for NPM registry malware

### PR DESCRIPTION
**Pull request by tstromberg**
Results of an initial audit to see how well we detect malware from the Backstabbers-Knife-Collection NPM repository. This adds enough rules to detect the first 20+ samples I reviewed, but there is still a long way to go.

While most of the improvements are focused on NPM and NodeJS, some rules are universally improved, such as the browser credential stealer rules.

NOTE: This also fixes a minor bug in `refresh-testdata.sh` which would corrupt any samples that had a `.json` extension.

### CODE REVIEW

File: rules/kernel/platform.yara [hunk 1]
```diff
@@ -67,9 +67,14 @@ rule npm_uname : medium {
...
+    $ = "process.platform"
+    $ = "process.arch"
+    $ = "process.versions"
+    $ = "os.platform()"
+    $ = "os.arch()"
+    $ = "os.release()"
   condition:
     any of them
 }
+
```
Comment (egibs) [inline rules/kernel/platform.yara:79]: Nit: two empty lines here

File: rules/user/lookup.yara [hunk 2]
```diff
@@ -1,14 +0,0 @@
-rule getpwuid : harmless {
```
Comment (egibs) [inline rules/user/lookup.yara:1]: Were these rules supposed to be added to `rules/user/current.yara`?

File: rules/random/bytes.yara [hunk 3]
```diff
@@ -0,0 +1,10 @@
+
+rule rand {
+	meta:
+		description = "generates random bytes"
+	strings:
+		$ref = ".randomBytes(" fullword
+	condition:
+		$ref
+}
+
```
Comment (egibs) [inline rules/random/bytes.yara:10]: Nit: empty line

File: rules/combo/stealer/browser.yara [hunk 4]
```diff
@@ -21,26 +21,39 @@ rule multiple_browser_credentials : high {
...
+	$fs_user_data = "User Data"
+
+	$name_brave = "Brave-Browser"
+    $name_brave_software = "BraveSoftware"
+    $name_chrome = "Google Chrome"
+    $name_opera_gx = "OperaGX"
+	$name_firefox = "Firefox"
+	$name_opera_gx_dir = "Opera GX"
+    $name_opera = "Opera Software"
+	$name_yandex = "YandexBrowser"
```
Comment (egibs) [inline rules/combo/stealer/browser.yara:51]: Would it make sense to add Safari or Vivaldi here (DDG and Orion are also possibilities)?

File: rules/combo/recon/nodejs.yara [hunk 5]
```diff
@@ -1,14 +1,53 @@
...
     $proc3 = "process.versions"
-    $h = "http.request"
+	$request_http = /require\([\"\']https{0,1}[\"\']\)/
     $post = "POST"
   condition:
-    filesize < 33554432 and $h and $post and any of ($proc*)
+    filesize < 16KB and $request_http and $post and any of ($proc*)
+}
+
+rule nodejs_phone_home : high {
```
Comment (egibs) [inline rules/combo/recon/nodejs.yara:16]: Does this need to be private or does that not matter all that much?

File: rules/combo/stealer/browser.yara (hunk 4 above)
Comment (tstromberg) [reply rules/combo/stealer/browser.yara:51]: Safari is special as it cookies in `~/Library/Cookies` (covered by $f_cookies)

I haven't seen malware samples that target the others, but if I do, we'll add them here.

File: rules/combo/recon/nodejs.yara (hunk 5 above)
Comment (tstromberg) [reply rules/combo/recon/nodejs.yara:16]: nope, I still want this rule to show if it matches, even if nothing that relies on it does.

File: rules/user/lookup.yara (hunk 2 above)
Comment (tstromberg) [reply rules/user/lookup.yara:1]: good catch that these were accidentally removed - re-added them.

**Review by egibs — APPROVED** [review]

**Pull request merged**